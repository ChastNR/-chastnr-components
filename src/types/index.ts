export * from "./input";
export * from "./optionRenderComponent";
export * from "./searchOption";
