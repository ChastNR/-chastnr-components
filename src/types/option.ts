export interface IOption<T = string | number> {
  className?: string;
  disabled?: boolean;
  title: React.ReactNode;
  value: T;
}
