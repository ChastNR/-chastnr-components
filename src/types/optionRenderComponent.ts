import { PropsWithChildren } from "react";

interface OptionRenderComponentProps {
  className?: string;
  label: string;
  onClick?: () => void;
}

export type OptionRenderComponent = (
  props: PropsWithChildren<OptionRenderComponentProps>
) => JSX.Element;
