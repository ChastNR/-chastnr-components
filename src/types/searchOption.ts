export interface SearchOption {
  label: string;
  value: string | number;
}
