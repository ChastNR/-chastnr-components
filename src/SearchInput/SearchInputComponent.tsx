import { IInputProps, Input } from '../Input';
import { OptionsContainer } from '../Option';
import { ISearchOption } from '../types';

interface ISearchInputProps extends IInputProps {
  isOptionsAvailable?: boolean;
  options?: ISearchOption[];
  renderOption?: (option: ISearchOption, index: number) => React.ReactNode;
}

export const SearchInputComponent: React.FC<ISearchInputProps> = ({
  content,
  isOptionsAvailable,
  options = [],
  renderOption,
  ...restProps
}) => {
  const showOptions = (isOptionsAvailable || options.length > 0) && !!renderOption;

  return (
    <Input
      {...restProps}
      content={
        <>
          {content}
          {showOptions && <OptionsContainer>{options.map(renderOption)}</OptionsContainer>}
        </>
      }
    />
  );
};
