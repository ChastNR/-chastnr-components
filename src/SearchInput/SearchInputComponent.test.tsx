import { fireEvent, render } from '@testing-library/react';
import { ISearchOption } from '../types';
import { SearchInputComponent } from './SearchInputComponent';

describe('SearchInputComponent', () => {
  const testOptions: ISearchOption[] = [{ label: 'test', value: 'test' }];
  const mockOnChange = jest.fn();
  const mockRenderOption = jest.fn();

  it('should render properly', () => {
    const searchInputComponent = render(
      <SearchInputComponent
        label="testLabel"
        options={testOptions}
        value=""
        onChange={mockOnChange}
        renderOption={mockRenderOption}
      />
    );

    expect(searchInputComponent.container.firstChild).toMatchSnapshot();
  });

  it('should show options container', () => {
    const searchInputComponent = render(
      <SearchInputComponent
        label="testLabel"
        value=""
        isOptionsAvailable
        onChange={mockOnChange}
        renderOption={mockRenderOption}
      />
    );

    const selectOptionsElement =
      searchInputComponent.container.getElementsByClassName('slt__opts')[0];

    expect(selectOptionsElement).toBeDefined();
  });
});
