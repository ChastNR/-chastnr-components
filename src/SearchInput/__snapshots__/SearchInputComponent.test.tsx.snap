// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInputComponent should render properly 1`] = `
<div
  class="ctrl"
>
  <div
    class="ctrl__cntr"
    role="textbox"
    tabindex="0"
  >
    <label
      class="ctrl__lb"
    >
      testLabel
    </label>
    <input
      autocomplete="off"
      type="text"
      value=""
    />
    <div
      class="slt__opts"
    />
  </div>
  <div
    class="er__cntr"
    style="height: 0px;"
  >
    <div />
  </div>
</div>
`;
